docs/content/examples/grid/SearchingAndFiltering.js

```diff
@@ -0,0 +1,387 @@
+                        <Tab value-bind="$page.code1.tab" tab="grid" mod="code" text="Grid" default />
+                    </div>
+
+                    <CodeSnippet visible-expr="{$page.code1.tab}=='controller'" fiddle="xLA9BNuv">{`
+                        class PageController extends Controller {
+                            onInit() {
+                                this.store.set(
+                                    "$page.employees",
+                                    Array.from({ length: 10 }).map((r, i) => ({
+                                        id: i + 1,
+                                        fullName: casual.full_name,
+                                        phone: casual.phone,
+                                        city: casual.city
+                                    }))
+                                );
+                            }
+                        }
+                    `}</CodeSnippet>
+                    <CodeSnippet visible-expr="{$page.code1.tab}=='grid'" fiddle="xLA9BNuv">{`
+                        <TextField
+                            value-bind="$page.search"
+                            icon="search"
+                            placeholder="Search..."
+                        />
+                        <Grid
+                            records-bind="$page.employees"
+                            columns={[
+                                { header: "Name", field: "fullName" },
+                                { header: "Phone", field: "phone" },
+                                { header: "City", field: "city" }
+                            ]}
+                            emptyText="No records match the specified filter"
+                            filterParams-bind="$page.search"
+                            onCreateFilter={(search) => {
+                                if (!search) return () => true;
+                                let predicate = getSearchQueryPredicate(search);
+                                return (record) =>
+                                    predicate(record.fullName) ||
+                                    predicate(record.phone) ||
+                                    predicate(record.city);
+                            }}
+                            scrollable
+                            style={{ height: "315px" }}
+                        />
+                    `}</CodeSnippet>
+                </Content>
+            </CodeSplit>
+
+            ## Searching Tree Grid
+            `Tree Grid` also supports searching. While traversing the tree, we need
+            to separate two cases: leaf nodes and non-leaf nodes. Leaf nodes can be
+            checked immediately and added to the result if they match the filter.
+            For non-leaf nodes, we must check their children too.
+
+            <CodeSplit>
+                <div class="widgets">
+                    <div style="margin-bottom: 10px; width: 100%">
+                        <TextField
```

Comment:
It would be better to use a separate variable for searching the tree, i.e. `$page.treeSearch` or `$page.treeQuery`.